smart-home-editor-extension/src/browser/smart-home-editor-frontend-module.ts

```diff
@@ -4,31 +4,74 @@
 
 import { SmartHomeEditorCommandContribution, SmartHomeEditorMenuContribution } from './smart-home-editor-contribution';
 import {
-    CommandContribution,
-    MenuContribution
+  CommandContribution,
+  MenuContribution
 } from "@theia/core/lib/common";
 import { ContainerModule } from "inversify";
-import { WidgetFactory } from "@theia/core/lib/browser";
-import { ResourceProvider } from "@theia/core/lib/common";
-import { TreeEditorWidget, TreeEditorWidgetOptions } from "theia-tree-editor";
+import { WidgetFactory, OpenHandler } from "@theia/core/lib/browser";
+import { ResourceProvider, Resource } from "@theia/core/lib/common";
+import { ResourceSaveable, TreeEditorWidget, TreeEditorWidgetOptions, TheiaTreeEditorContribution } from "theia-tree-editor";
 import URI from "@theia/core/lib/common/uri";
-import App, {initStore} from "../App";
+import App, { initStore } from "../App";
+import { ThemeService } from '@theia/core/lib/browser/theming';
+
+import { getData } from "@jsonforms/core";
+
+const LIGHT_THEME_ID = "light"
+
+class MyResourceSaveable extends ResourceSaveable {
+  constructor(resource: Resource, getData: () => any) {
+    super(resource, getData);
+  }
+  onSave(data: any) {
+    return postRequest('http://localhost:9091/services/convert/json', JSON.stringify(data), 'application/json')
```

Comment:
Use MessageService to report an error if the promise is rejected. You can get the `MessageService` injected in the contructor.